src/App.js

```diff
@@ -14,11 +15,106 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const options = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+class App extends React.Component {
+  state = {
+    isVisible: false,
+    isReversed: false,
+    sortBy: '',
+    minLength: 1,
+  }
+
+  start = () => {
+    this.setState(state => ({
+      isVisible: !state.isVisible,
+    }));
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  sortByName = () => {
+    this.setState({
+      sortBy: 'name',
+    });
+  }
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'length',
+    });
+  }
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+      minLength: 1,
+    });
+  }
+
+  select = ({ target }) => {
+    this.setState({
+      minLength: +target.value,
+    });
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <h1>Goods</h1>
+        <button
+          type="button"
+          onClick={this.start}
+        >
+          {this.state.isVisible ? 'Hide' : 'Start'}
+        </button>
+
+        <br />
+
+        <button
+          type="button"
+          onClick={this.reverse}
+        >
+          Reverse
+        </button>
+
+        <button
+          type="button"
+          onClick={this.sortByName}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={this.sortByLength}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={this.reset}
+        >
+          Reset
+        </button>
+
+        <br />
+
+        <select value={this.state.minLength} onChange={this.select}>
+          {options.map(x => <option key={x}>{x}</option>)}
+        </select>
+
+        {this.state.isVisible
+          && <GoodsList goodsList={goodsFromServer} {...this.state} />}
```

Comment:
don't pass all state into GoodsList, build your code in that way, which allow you to pass already prepared goods into goodsList component